release-notes-agent/src/main/java/com/epam/release/notes/agent/service/action/mail/EmailTemplateNotificationService.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ */
+package com.epam.release.notes.agent.service.action.mail;
+
+import com.epam.release.notes.agent.entity.github.GitHubIssue;
+import com.epam.release.notes.agent.entity.jira.JiraIssue;
+import com.epam.release.notes.agent.entity.mail.EmailContent;
+import com.epam.release.notes.agent.entity.version.Version;
+import com.epam.release.notes.agent.service.version.ApplicationVersionService;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring5.SpringTemplateEngine;
+
+import java.util.List;
+import java.util.Locale;
+
+@Service
+public class EmailTemplateNotificationService implements TemplateNotificationService {
+
+    private static final String OLD_VERSION = "oldVersion";
+    private static final String NEW_VERSION = "newVersion";
+    private static final String JIRA_ISSUES = "jiraIssues";
+    private static final String GITHUB_ISSUES = "gitHubIssues";
+    private static final String EXCEPTION_MESSAGE = "This kind of version status is not handled in current method";
+
+    private final SpringTemplateEngine templateEngine;
+    private final ApplicationVersionService applicationVersionService;
+    private final String emailToAdminTemplateName;
+    private final String emailToSubscribersTemplateName;
+    private final String emailToSubscribersWithoutIssuesTemplateName;
+    private final String emailToAdminTitle;
+    private final String emailToSubscribersTitle;
+
+    public EmailTemplateNotificationService(
+            final SpringTemplateEngine templateEngine,
+            final ApplicationVersionService applicationVersionService,
+            @Value("${release.notes.agent.name.of.admin.email.template}") final String emailToAdminTemplateName,
+            @Value("${release.notes.agent.name.of.subscribers.email.template}") final String emailToSubscribersTemplateName,
+            @Value("${release.notes.agent.name.of.subscribers.email.without.issues.template}")
+            final String emailToSubscribersWithoutIssuesTitle,
+            @Value("${release.notes.agent.email.to.admin.subject}") final String emailToAdminTitle,
+            @Value("${release.notes.agent.email.to.subscribers.subject}") final String emailToSubscribersTitle
+    ) {
+        this.templateEngine = templateEngine;
+        this.applicationVersionService = applicationVersionService;
+        this.emailToAdminTemplateName = emailToAdminTemplateName;
+        this.emailToSubscribersTemplateName = emailToSubscribersTemplateName;
+        this.emailToAdminTitle = emailToAdminTitle;
+        this.emailToSubscribersTitle = emailToSubscribersTitle;
+        this.emailToSubscribersWithoutIssuesTemplateName = emailToSubscribersWithoutIssuesTitle;
+    }
+
+    @Override
+    public EmailContent populate(final String oldVersion,
```

Comment:
why you didn't change this method signature as discussed? by moving all parameters to separate object? is there any problem with that?